cmd/av/stack_switch.go

```diff
@@ -99,6 +117,52 @@ func stackSwitchBranchList(repo *git.Repo, tx meta.ReadTx, branches map[string]*
 	return ret
 }
 
+func parseBranchName(input string) (string, error) {
+	if branch, err := parsePullRequestURL(input); err == nil {
+		return branch, nil
+	}
+
+	return input, nil
+}
+
+var PULL_REQUEST_URL_REGEXP = regexp.MustCompile(`^/([^/]+)/([^/]+)/pull/(\d+)`)
+
+func parsePullRequestURL(prURL string) (string, error) {
+	u, err := url.Parse(prURL)
+	if err != nil {
+		return "", errors.Wrap(err, "failed to parse URL")
+	}
+
+	if u.Scheme != "https" && u.Scheme != "http" {
+		return "", errors.New("URL is not a pull request URL")
+	}
+
+	m := PULL_REQUEST_URL_REGEXP.FindStringSubmatch(u.Path)
+	if m == nil {
+		return "", errors.New(fmt.Sprintf("URL is not a pull request URL format:%s", prURL))
+	}
+
+	client, err := getGitHubClient()
+	if err != nil {
+		return "", errors.Wrap(err, "failed to get GitHub client")
+	}
+
+	prID, err := strconv.Atoi(m[3])
+	if err != nil {
+		return "", errors.Wrap(err, "failed to parse pull request ID")
+	}
+	pr, err := client.GetPullRequest(context.Background(), gh.PullRequestOpts{
+		Owner:  m[1],
+		Repo:   m[2],
+		Number: int64(prID),
+	})
```

Comment:
You can find the branch based on https://github.com/aviator-co/av/blob/d74f1e1103e0e7e2eef3f8f3c7a3a08c38e81aa2/internal/meta/branch.go#L22. This shouldn't query GitHub.

Reply:
Does this mean that this [AllBranches method](https://github.com/aviator-co/av/pull/360/files#diff-1421442c9c429b4ddaaf38864ee227f4960ec5be73dcbcbd2031f69f92acfef2R148) also holds remote branch information locally?

Reply:
fixed https://github.com/aviator-co/av/pull/360/commits/6f8cc4a349f1850dd71cd0fea9ff385db37bbde8

Reply:
The local metadata DB should keep those PR info.